openpgp/read.go

```diff
@@ -443,7 +446,10 @@ func (scr *signatureCheckReader) Read(buf []byte) (int, error) {
 	}
 
 	if sensitiveParsingError != nil {
-		return n, errors.StructuralError("parsing error")
+		if scr.md.decrypted != nil {
```

Comment:
I'd add comments where we do this sort of check to explain the high-level logic a bit for someone unfamiliar with the lib 👼